src/index.tsx

```diff
@@ -27,6 +33,9 @@ interface BuiltInValidationAttrs {
   maxLength?: number | BuiltInValidationAttrsFunction<number>;
   min?: number | BuiltInValidationAttrsFunction<number>;
   max?: number | BuiltInValidationAttrsFunction<number>;
+  // TODO: Pattern not valid on select/textarea - can we enforce this via types?
+  // Set up a discriminated union in InputDefinition based on element:'select'?
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern
```

Comment:
I think what you're looking for is conditional types
https://www.typescriptlang.org/docs/handbook/2/conditional-types.html

Reply:
Got this handled with a new discriminated unions ([`InputValidationAttrs`](https://github.com/brophdawg11/remix-validity-state/pull/31/commits/fb7a5d02e97e87ff448750ba9a3d01583a98d68d#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405R76), [`ControlDefinition`](https://github.com/brophdawg11/remix-validity-state/pull/31/files#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405R124)) so we can now use provided `element`/`type` fields to ensure only valid attributes are defined ion `validationAttrs`

```ts
interface FormSchema {
  inputs: {
    name: InputDefinition;
    birthday: InputDefinition;
    state: SelectDefinition;
    bio: TextAreaDefinition;
  };
}

let formDefinition: FormSchema = {
  inputs: {
    name: {
      validationAttrs: {
        // Not specifying a `type` defaults to a text-like input (text, search, password, etc)
        required: true,
        minLength: 5,
      },
    },
    birthday: {
      validationAttrs: {
        // type:'date' further restricts to only the supportted validation attributes
        type: 'date',
        required: true,
        min: '1900-01-01',
      },
    },
    state: {
      // element: 'select' sets `validationAttrs` to only support `required`
      element: "select",
      validationAttrs: {
        required: true,
      },
    },
    bio: {
      // element: 'textarea' sets `validationAttrs` to only support `required`/`minLength`/`maxLength`
      element: "textarea",
      validationAttrs: {
        required: true,
        maxLength: 400,
      },
    },
  },
};
```